tempesta_fw/t/unit/test_hash.c

```diff
@@ -63,27 +63,26 @@ TEST(tfw_hash_str, calcs_diff_hash_for_diff_str)
 TEST(tfw_hash_str, calcs_same_hash_for_diff_chunks_n)
 {
 	unsigned long h1, h2, h3;
-
-	TfwStr s1 = { .len = 17, .ptr = (void *)"Host: example.com" };
-
-	TfwStr s2c1 = {	.len = 14, .ptr = (void *)"Host: example." };
-	TfwStr s2c2 = {	.len = 3, .ptr = (void *)"com" };
+	TfwStr s1 = { .len = 17, .data = "Host: example.com" };
+	TfwStr s2c1 = {	.len = 14, .data = "Host: example." };
+	TfwStr s2c2 = {	.len = 3, .data = "com" };
 	TfwStr s2chunks[] = { s2c1, s2c2 };
 	TfwStr s2 = {
+
 		.len = 14 + 3,
-		.ptr = s2chunks
+		.chunks = s2chunks
 	};
 
-	TfwStr s3c1 = {	.len = 1, .ptr = (void *)"H" };
-	TfwStr s3c2 = {	.len = 0, .ptr = (void *)"" };
-	TfwStr s3c3 = {	.len = 3, .ptr = (void *)"ost" };
-	TfwStr s3c4 = {	.len = 1, .ptr = (void *)":" };
-	TfwStr s3c5 = {	.len = 12, .ptr = (void *)" example.com" };
-	TfwStr s3c6 = {	.len = 0, .ptr = NULL };
+	TfwStr s3c1 = {	.len = 1, .data = "H" };
+	TfwStr s3c2 = {	.len = 0, .data = "" };
+	TfwStr s3c3 = {	.len = 3, .data = "ost" };
+	TfwStr s3c4 = {	.len = 1, .data = ":" };
+	TfwStr s3c5 = {	.len = 12, .data = " example.com" };
+	TfwStr s3c6 = {	.len = 0, .data = NULL };
```

Comment:
There are macros `DEFINE_TFW_STR()` and `TFW_STR_FROM()` that can be used for this purpose.